src/KafkaFlow/Event.cs

```diff
@@ -25,23 +25,27 @@ public IEventSubscription Subscribe(Func<TArg, Task> handler)
             return new EventSubscription(() => this.handlers.Remove(handler));
         }
 
-        internal async Task FireAsync(TArg arg)
+        internal Task FireAsync(TArg arg)
+        {
+            return Task.WhenAll(this.FireAsyncCall(arg));
+        }
+
+        private IEnumerable<Task> FireAsyncCall(TArg arg)
         {
             foreach (var handler in this.handlers)
             {
-                try
-                {
                     if (handler is null)
                     {
                         continue;
                     }
 
-                    await handler.Invoke(arg);
-                }
-                catch (Exception e)
-                {
-                    this.logHandler.Error("Error firing event", e, new { Event = this.GetType().Name });
-                }
+                    yield return handler.Invoke(arg).ContinueWith(t =>
```

Comment:
Please fix the indentation on this block (after the foreach the indentation has one extra level)